src/mscorlib/src/System/IntPtr.cs

```diff
@@ -109,7 +109,16 @@ public unsafe IntPtr(void* value)
     
         [System.Security.SecuritySafeCritical]  // auto-generated
         public unsafe override int GetHashCode() {
-            return unchecked((int)((long)m_value));
+#if FEATURE_CORECLR
+    #if WIN32
+                return unchecked((int)((long)m_value));
+    #else
+                long l = (long)m_value;
+                return (unchecked((int)l) ^ (int)(l >> 32));
+    #endif
+#else
+                return unchecked((int)((long)m_value));
```

Comment:
Nit: The indentation here is more than in `UIntPtr`. It'd be nice to be consistent.